### PR TITLE
Dls 4239

### PR DESCRIPTION
**Pull request by gavinharris43**

### CODE REVIEW

File: app/uk/gov/hmrc/softdrinksindustrylevy/connectors/DesHelpers.scala [hunk 1]
```diff
@@ -27,14 +27,16 @@ abstract class DesHelpers(servicesConfig: ServicesConfig) {
...
+
+  private def desHeaders = Seq("Environment" -> serviceEnvironment, "Authorization" -> serviceKey)
+
   def desPost[I, O](
     url: String,
     body: I)(implicit wts: Writes[I], rds: HttpReads[O], hc: HeaderCarrier, ec: ExecutionContext): Future[O] =
-    http.POST[I, O](url, body)(wts, rds, addHeaders, ec)
+    http.POST[I, O](url, body, headers = desHeaders)(wts, rds, addHeaders, ec)
 
   def addHeaders(implicit hc: HeaderCarrier): HeaderCarrier =
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/softdrinksindustrylevy/connectors/DesHelpers.scala:40]: remove this completely and change the post to just use the implicit hc from method call

File: app/uk/gov/hmrc/softdrinksindustrylevy/controllers/RegistrationController.scala [hunk 2]
```diff
@@ -146,20 +152,20 @@ class RegistrationController(
...
+      logger.info("TE returned SUCCEEDED for subscription")
       buffer.remove("subscription.utr" -> utr) map { _ =>
-        Logger.info("deleting record from the buffer and returning OK & subscription json")
+        logger.info("deleting record from the buffer and returning OK & subscription json")
         Ok(Json.toJson(s))
       }
     case a if a.state == "ERROR" =>
-      Logger.error(a.errorResponse.getOrElse("unknown tax enrolment error")) // TODO also deskpro
-      Logger.info("TE returned ERROR, returning OK and the subscription json ")
+      logger.error(a.errorResponse.getOrElse("unknown tax enrolment error")) // TODO also deskpro
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/softdrinksindustrylevy/controllers/RegistrationController.scala:161]: Remove the comment as we don't really know why it was there first time

File: build.sbt [hunk 3]
```diff
@@ -39,19 +36,19 @@ libraryDependencies ++= Seq(
   "org.jsoup"                   %  "jsoup"               % "1.13.1",
   "org.mockito"                 %  "mockito-core"        % "3.4.6",
```
Comment (calinursaciuc) [inline build.sbt:37]: try mockito core 3.11.1

File: project/build.properties [hunk 4]
```diff
@@ -1 +1 @@
-sbt.version=1.3.13
+sbt.version=1.4.9
```
Comment (calinursaciuc) [inline project/build.properties:1]: Try sbt 1.5.2 if we it doesn't throw startup or compile errors

File: project/plugins.sbt [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.0.0")
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "2.1.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "2.2.0")
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.0.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.1.0")
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "1.5.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "2.0.0")
```
Comment (calinursaciuc) [inline project/plugins.sbt:12]: remove artifactory and git versioning

**Review by calinursaciuc — COMMENTED** [review]
In DES connector all calls to use explicit headers and for implicit hc just use the one from the method

File: project/build.properties (hunk 4 above)
Comment (gavinharris43) [reply project/build.properties:1]: throws additional errors

File: build.sbt [hunk 6]
```diff
@@ -61,17 +58,13 @@ libraryDependencies ++= Seq(
...
-  "uk.gov.hmrc"               %% "mongo-lock"                    % "6.23.0-play-26",
-  "uk.gov.hmrc"               %% "simple-reactivemongo"          % "7.30.0-play-26",
-  "org.scala-stm"             %% "scala-stm"                     % "0.9.1"
-)
-
-resolvers ++= Seq(
-  Resolver.bintrayRepo("hmrc", "releases"),
-  Resolver.jcenterRepo
+  "org.typelevel"             %% "cats-core"                     % "2.4.0",
+  "uk.gov.hmrc"               %% "bootstrap-backend-play-26"     % "5.3.0",
```
Comment (rambabu-posa) [inline build.sbt:62]: We can use `5.4.0` for bootstrap-backend-play-26. They did one small improvement in this release.

**Review by calinursaciuc — APPROVED** [review]

**Pull request merged**